### DISCUSSION TITLE
Cooldown Swipe on Weak Auras not Overlaid by Masque

### DISCUSSION THREAD

**Discussion by raphaeloliveira** [SFX-WoW/Masque#149] [Questions] [answered]
### Game Flavor

Retail

### Game Version

9.1.0.40443

### Add-On Version

9.1.0

### Is the improvement related to a problem?

Yes, I'm using `Masque` to customise my `Weak Auras` and if I use the `Cooldown Swipe` option, the swipe is not masked by `Masque` as per screenshot below. You can see the pointy square edges.

### Describe the Improvement

I would love if the cooldown swipe would also be masked by `Masque`.

### Screenshots

![image](https://user-images.githubusercontent.com/2700131/136763377-9620a319-7831-4e06-abf2-ad4439062fbd.png)
![image](https://user-images.githubusercontent.com/2700131/136763513-6e605e81-c047-4a4b-af08-b3077e25f072.png)

Thank you!

**Comment by raphaeloliveira** [comment]
I also asked for support on the `Weak Auras` discord and they said they thought `Masque` would already do that. However, they also gave me this snippet https://discord.com/channels/172440238717665280/218084462523711489/897044939802804255 which creates a circular swipe. Doesn't quite match as well the square shape with rounded corners but I thought I would share.

![image](https://user-images.githubusercontent.com/2700131/136765440-44616626-bc82-48c9-9cc4-67c78ca31550.png)

**Comment by StormFX** [comment]
This actually isn't an issue with Masque or WeakAuras, but rather the skin you're using. The skin itself should provide the mask texture for the Cooldown swipe. If you don't mind my asking, what skin is it that you're using?

Edit: Also, if you really want to use that skin, I could probably explain how to add something that looks better, assuming you have an image editor and don't mind a bit of code editing.

**Reply by raphaeloliveira** [reply]
Hey @StormFX, I'm using Renaitre (https://www.curseforge.com/wow/addons/masque-renaitre) which unfortunately doesn't seem to be updated anymore but it is one of the most cool looking skins in my opinion.

> Edit: Also, if you really want to use that skin, I could probably explain how to add something that looks better, assuming you have an image editor and don't mind a bit of code editing.

Yes, please, if you can help me how to fix or workaround that it would be much appreciated!

Thank you!

**Comment by StormFX** [comment, marked as answer]
Ok. It looks like you're using the _Beveled_ skin. If not, the same applies to whichever skin you're using, just with a different path. First, open `Skins.lua` with a text editor. Find the `Cooldown` section of the skin you want to change, for example, for _Beveled_, it's lines `30`-`33`:

```lua
	Cooldown = {
		Width = 30,
		Height = 30,
	},
```

Change it to something like:

```
	Cooldown = {
		Texture = [[Interface\AddOns\Masque_Renaitre\Textures\Beveled\Overlay]],
		Color = {0, 0, 0, 0.8},
		Width = 30,
		Height = 30,
		Point = "CENTER",
		RelPoint = "CENTER",
		OffsetX = 0,
		OffsetY = 0,
	},
```

Note that you'll likely need to adjust the `Height` and `Width` attributes to fit it properly, but you can change and save it, then reload the interface to test it while in-game.

The other thing you may want to do is use a different image, since the `Overlay` image has a skull in the center. In that case, just open the image in an image editor and cover the skull with white. Then, save the image under a different name, like `Overlay2` and then update the above code to point to the new image.

**Reply by raphaeloliveira** [reply]
Thank you so much @StormFX, it worked perfectly!!!

**Reply by StormFX** [reply]
You bet! :)
